gidgethub/sansio.py

```diff
@@ -378,6 +380,7 @@ def format_url(url: str, url_vars: Mapping[str, Any]) -> str:
 
     The dict provided in url_vars is used in URI template formatting.
     """
-    url = urllib.parse.urljoin(DOMAIN, url)  # Works even if 'url' is fully-qualified.
+    base_url = base_url or DOMAIN
```

Comment:
I think it's safer to have classic `None`-check here.

Reply:
I added the check `if base_url is None`

Reply:
@webknjaz safer how? The empty string isn't valid either so there's no need to differentiate between `''` and `None`. And if you have the base URL as configurable then you will be passing it in explicitly anyway.

But I argue the default be set on the parameter and then we be done with it. If people are worried about an incorrect type being passed in or the exception not being obvious then I would argue they should use mypy to catch it.

Reply:
Fair enough. I was just thinking that I'd prefer a more explicit check because it seems to be a more idiomatic "if default" check. But you are right, maybe it doesn't matter much in this case.